Exec/unit_tests/diffusion_test/problem_bc_fill.H

```diff
@@ -33,15 +33,37 @@ void problem_bc_fill(int i, int j, int k,
     const Real* problo = geomdata.ProbLo();
 
     Real r[3] = {0.0_rt};
-    r[0] = problo[0] + dx[0] * (static_cast<Real>(i) + 0.5_rt) - problem::center[0];
-
+    r[0] = problo[0] + dx[0] * (static_cast<Real>(i) + 0.5_rt);
 #if AMREX_SPACEDIM >= 2
-    r[1] = problo[1] + dx[1] * (static_cast<Real>(j) + 0.5_rt) - problem::center[1];
+    r[1] = problo[1] + dx[1] * (static_cast<Real>(j) + 0.5_rt);
+#endif
+#if AMREX_SPACEDIM == 3
+    r[2] = problo[2] + dx[2] * (static_cast<Real>(k) + 0.5_rt);
 #endif
+    if (coord <= 1) {
+        for (int i; i < AMREX_SPACEDIM; ++i) {
+            r[i] -= problem::center[i];
+        }
+    }
 
+    if (diffuse_use_amrex_mlmg) {
```

Comment:
can you add a comment here explaining the logic

Reply:
done